bin/libast_prereq.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/sh
+
+# This script is used for feature detection and setting up other configurations that are required to build libast
+set -x
+set -e
+
+# This script is run from an unspecified directory so we have to determine directory paths
+# http://mesonbuild.com/Reference-manual.html#run_command
+script_path=`realpath $0`
+bin_dir=`dirname $script_path`
+base_dir=`dirname $bin_dir`
+
+PATH=$bin_dir:$PATH
+
+# Legacy build system keeps files generated by iffe with different names under different directories
+# Each array listed below is processed separately and headers generated by iffe are kept in separate directories
+# in a way that is compatible with legacy build system
+iffe_tests=( 'api' 'aso' 'ccode' 'common' 'float' 'fs' 'iconv' 'lib' 'locale' 'mmap' 'ndbm' 'options' 'prog' 'sched' 'sfio' 'siglist' 'sizeof' 'spawn' 'standards' 'sys' 'sys_calls' 'time' 'tmlib' 'tty' 'tv' 'tvlib' 'vfork' 'vmalloc' 'wait')
+
+iffe_tests_2=( 'tv' 'tmx' )
+
+iffe_c_tests=('align.c' 'fcntl.c'  'limits.c'  'map.c'  'mode.c') 
+
+iffe_c_tests_2=('atdev.c' 'sfinit.c' 'signal.c')
+
+iffe_c_tests_3=('align.c')
+
+iffe_sh_tests=('preroot.sh'  'sig.sh')
+
+iffe_sh_tests_2=('param.sh' )
+
+iffe_sh_tests_3=('libpath.sh' )
+
+pushd $base_dir/src/lib/libast/features
+
+iffe -v -X ast -X std -c 'cc -D_BLD_DLL -D_BLD_ast    ' run param.sh
+mv $base_dir/src/lib/libast/features/FEATURE/param.sh $base_dir/src/lib/libast/features/FEATURE//param
```

Comment:
s,//,/,

Reply:
Also, pretty much everywhere else in this script you doublequote paths like these. It's unlikely `$base_dir` will contain problematic chars (e.g., a space) but better to be safe and quote the strings.